sdk/storage/azure-storage-stress/src/main/java/com/azure/storage/stress/TelemetryHelper.java

```diff
@@ -195,9 +196,22 @@ private void trackCancellation(Instant start, Span span) {
     private void trackFailure(Instant start, Throwable e, Span span) {
         Throwable unwrapped = Exceptions.unwrap(e);
 
+        // Check if the message contains "NativeIoException" and the unwrapped exception is not already a NativeIoException
+
+        // Check if the unwrapped exception is a RuntimeException
+        if (unwrapped instanceof RuntimeException) {
+            String message = unwrapped.getMessage();
+            if (message.contains("NativeIoException")) {
+                unwrapped = new io.netty.channel.unix.Errors.NativeIoException("recvAddress", Errors.ERRNO_ECONNRESET_NEGATIVE);
+            } else if (message.contains("TimeoutException")) {
+                unwrapped = new TimeoutException(message);
+            }
+            // may need to add more known exceptions here
```

Comment:
We can probably remove this comment. Reading it, it was unclear if there were more exceptions we need to map or if it is just a pointer to where to update the logic for the next time.